module/zfs/brt.c

```diff
@@ -1420,7 +1420,7 @@ brt_pending_entry_compare(const void *x1, const void *x2)
 	const blkptr_t *bp1 = &bpe1->bpe_bp, *bp2 = &bpe2->bpe_bp;
 	int cmp;
 
-	cmp = TREE_CMP(BP_PHYSICAL_BIRTH(bp1), BP_PHYSICAL_BIRTH(bp2));
+	cmp = TREE_CMP(BP_GET_BIRTH(bp1), BP_GET_BIRTH(bp2));
```

Comment:
I see in number of places you are replacing BP_PHYSICAL_BIRTH() with BP_GET_BIRTH(), that implies additional logic, checking for embedded and zero.  This function is definitely not where we want to extra logic.  Generally unified BP_GET_BIRTH() without logical/physical separation looks confusing to me.

BTW, this chunk would conflict with one PR of mine that is currently open.

Reply:
I'm sorry, I haven't understood that you just renamed BP_PHYSICAL_BIRTH() to BP_GET_BIRTH(), so scratch the part of checks.  Though the name is still confusing.

Reply:
BP_GET_BIRTH is now the complement to BP_SET_BIRTH. Before it was BP_PHYSICAL_BIRTH and BP_SET_BIRTH which was confusing. I think many confused BP_PHYSICAL_BIRTH with getting the physical birth which it does not always do.

Reply:
BP_SET_BIRTH() sets both times, so the only confusion I see is why not use separately BP_SET_PHYSICAL_BIRTH() and BP_SET_LOGICAL_BIRTH().  Same time BP_GET_BIRTH() is less obvious.

Reply:
It was designed to make it easier for the developer that isn't familiar with dedup and doesn't know which birth time to set. So they are just convenience methods.